### DISCUSSION TITLE
Release on CRAN

### DISCUSSION THREAD

**Discussion by BerriJ** [ManuelHentschel/vscDebugger#170] [Ideas]
Hi,

first of all: thank you so much for this package.

I wonder if it's possible to release this on CRAN. I personally use the [RStudio Package Manager](https://packagemanager.rstudio.com/client/#/repos/1/overview) to keep track of package versions. In essence, RSPM creates CRAN snapshots and freezes them.

Other points to think about:

Pro: 
- CRAN may improve the visibility of this package

Cons:
- Complying with CRAN rules likely creates some initial workload (mainly creating / editing docs)

Of course, I'm happy to help to prepare the release:

Running devtools::check(cran = TRUE) throws several notes and warnings. I can easily resolve some of them. Do you want me to create a PR for this?

I'm looking forward to a fruitful discussion.

**Comment by eitsupi** [comment]
My understanding is that VSCode-R-Debugger (which is usually always updated to the latest version by VSCode) requires the same version of `vscDebugger`, so it would make sense to run `remotes::install_github("ManuelHentschel/vscDebugger@*release")` to install the latest release version from GitHub, instead of installing it from RSPM.

**Reply by eitsupi** [reply]
By the way, when I wrote https://github.com/microsoft/vscode-dev-containers/pull/954#issuecomment-886068895, I did not understand that `install_github` has a feature to install from the latest release.
After learning about this feature, I'm convinced that installing from GitHub is the best way to go, and I hope to add the `vscDebugger` to the Dockerfile in the next update. (The base image of the Dockerfile, `rocker/r-ver`, installs packages from RSPM by default)

**Reply by eitsupi** [reply]
I submitted a PR https://github.com/microsoft/vscode-dev-containers/pull/1007

**Comment by renkun-ken** [comment]
I think releasing vscDebugger to CRAN could make it difficult to update the vscode extension since any update might require a development version of the R package, making the CRAN release useless.

**Comment by BerriJ** [comment]
Thank you very much for your comments. It makes sense to me to keep it GitHub-Only and also thanks @eitsupi for mentioning remotes::install_github("ManuelHentschel/vscDebugger@*release"). I'm also maintaining a dev container and will use this feature from now on.

**Comment by eitsupi** [comment]
I thought it would be fine to install from GitHub, but forgot that this command `remotes::install_github("ManuelHentschel/vscDebugger@*release")` was limited by the rate limit of the GitHub API.
In other words, if you access the GitHub API in large numbers for other purposes, the installation will fail.

How about using R-universe?
https://www.r-bloggers.com/2021/06/how-to-create-your-personal-cran-like-repository-on-r-universe/

**Comment by ManuelHentschel** [comment]
Hi! Sorry, I totally didn't notice this thread, because apparently I was not "watching" the discussions here by default. Apologies for this super late response.

Thanks for your feedback and the suggestion! So far I was a bit hesitant about CRAN for various reasons
* As @renkun-ken mentioned, the vscode-R-Debugger extension and this package are coupled quite closely, and the release process on CRAN might make it difficult to keep the two in sync. 
* I haven't had much time recently to work on this, and there is some code cleanup and bug fixing I'd like to do first
* I am hoping to improve the DAP-server by using a separate thread in C/C++. This would ideally make it possible to run the debug adapter in the background of a normal R session (I have the impression that this is one of the most annoying aspects of using the debugger). This might entail some changes to the way vscDebugger and the vscode extension communicate, though.
* Before publishing on CRAN it might make sense to move this repo to REditorSupport

In general, though, I think it would make sense to have this package on CRAN at some point. I hope I'll have a bit more time for programming after the holidays and will try to address the points mentioned above then!
